### DISCUSSION TITLE
Personal Access Tokens

### DISCUSSION THREAD

**Discussion by soloam** [firefly-iii/firefly-iii#9233] [Firefly III]
Hello, I would like a clarification! Are the Personal Access Tokens temporary? Will they expire? I'm accessing the api by Personal Access Tokens and they keep getting unauthorized after some time, are they temporary? And if so, is it possible to make them last forever?

Thank you

**Comment by JC5** [comment]
Hey, great question!

They are long lived but I'm not sure how long. At least 90 days but I owe you the real answer.

**Reply by soloam** [reply]
So a time limit exists! So I have to update my home assistant integration to use OAuth2, that is the only way to ensure that users can keep connected without having to make new tokens!

**Reply by JC5** [reply]
Yes, but my own token has been valid for like a year now?

**Reply by gitnuu** [reply]
It may be valid for a year. It was created one year ago today at noon. I can't use it properly this afternoon. The current version is 6.1.19. During this time, upgrades were made through docker compose.

When accessed through [Personal Access Tokens], nginx displays {" GET/app/v1/about HTTP/1.1 "401}

[Non-English, translated by software]

**Comment by JC5** [comment]
I looked it up. They expire after one year.
